### PR TITLE
feat: user send message through channel

### LINKED ISSUE #79: User interaction: Send message through channel
- [x] Introduce probability for users to send messages in channels (or reuse existing)

### PR DESCRIPTION
**Pull request by lauti7**
This PR: #79

### CODE REVIEW

File: src/user.rs [hunk 1]
```diff
@@ -421,10 +454,16 @@ fn pick_random_action(probability_to_act: usize, channels_enabled: bool) -> Soci
             SocialAction::JoinChannel
         } else if rng.gen_ratio(1, 25) {
             SocialAction::UpdateStatus
-        } else if rng.gen_ratio(1, 3) {
+        } else if rng.gen_ratio(1, 10) {
             SocialAction::AddFriend
+        } else if channels_enabled {
+            if rng.gen_ratio(1, 5) {
```
Comment (moliva) [inline src/user.rs:460]: I think this can be inlined with the `if` in the line above and dedup the send message to DMs.

File: src/user.rs [hunk 2]
```diff
@@ -421,10 +454,16 @@ fn pick_random_action(probability_to_act: usize, channels_enabled: bool) -> Soci
             SocialAction::JoinChannel
         } else if rng.gen_ratio(1, 25) {
             SocialAction::UpdateStatus
-        } else if rng.gen_ratio(1, 3) {
+        } else if rng.gen_ratio(1, 10) {
```
Comment (moliva) [inline src/user.rs:457]: Out of curiosity, why do we want to change this?

File: src/user.rs [hunk 3]
```diff
@@ -439,6 +478,19 @@ async fn pick_random_room(rooms: &RwLock<Vec<OwnedRoomId>>) -> Option<OwnedRoomI
         .map(|room| room.to_owned())
 }
 
+async fn pick_random_channels(channels: &RwLock<HashSet<OwnedRoomId>>) -> Option<OwnedRoomId> {
+    let channels = channels.read().await;
+    if !channels.is_empty() {
```
Comment (moliva) [inline src/user.rs:483]: No need for this check, `choose` will automatically return `None` if the `slice/Vec` is empty

File: src/user.rs (hunk 2 above)
Comment (lauti7) [reply src/user.rs:457]: nope, it was just to keep the descending order but anyway I rolled it back.

File: src/user.rs (hunk 1 above)
Comment (lauti7) [reply src/user.rs:460]: yes u r right

**Review by moliva — APPROVED** [review]
LGTM.

**Pull request merged**